JsonSchema.Generation/SchemaGeneratorConfiguration.cs

```diff
@@ -34,11 +35,42 @@ public class SchemaGeneratorConfiguration
 	/// <remarks>
 	/// This can be replaced with any `Func&lt;string, string&gt;`.
 	/// </remarks>
-	public PropertyNamingMethod PropertyNamingMethod
+	[Obsolete($"Use {nameof(PropertyNameResolver)} instead.")]
+	public PropertyNamingMethod? PropertyNamingMethod
 	{
-		get => _propertyNamingMethod ??= PropertyNamingMethods.AsDeclared;
+		get => _propertyNamingMethod;
 		set => _propertyNamingMethod = value;
 	}
+
+	/// <summary>
+	/// Gets or sets the property name resolving method. Default is <see cref="PropertyNameResolvers.ByJsonPropertyName"/>.
+	/// </summary>
+	/// <remarks>
+	/// This can be replaced with any `Func&lt;MemberInfo, string&gt;`.
+	/// </remarks>
+	public PropertyNameResolver PropertyNameResolver
+	{
+		get
+		{
+			// upwards compatibility: this field was actively set, this is only used logic.
+			if (_propertyNameResolver is PropertyNameResolver resolver)
+			{
+				return resolver;
+			}
+
+			// backwards compatibility: if the old property was actively set in existing code existing code, code will use it
+			if (PropertyNamingMethod is PropertyNamingMethod method)
+			{
+				return new PropertyNameResolver(x => method(x.Name));
+			}
+
+			// if neither property was actively set, the behavior is as it was before. Prefer JsonPropertyName and Fallback to AsDeclared
+			return PropertyNameResolvers.ByJsonPropertyName;
```

Comment:
The attribute should override this setting completely.  If someone has explicitly stated the name they want to use in the JSON, then the generator needs to use that, regardless of what this setting is.  This shouldn't be part of the setting.

The default here needs to be `AsDeclared`, like it is currently for `PropertyNamingMethod`.

Reply:
Actually, the way i implemented it is exactly the current behavior:

In the current main it reads the name via PropertyNamingMethod which has AsDeclard as default.
And then if there is a JsonPropertyNameAttribute, it overrides the name with it.

Which is what I replaces a more flexible/mix-and-match/lego-brick way.

![image](https://github.com/gregsdennis/json-everything/assets/11560817/2822fb2f-5404-4daf-bf80-5b2221b09718)

Reply:
I mean, you'll always need a fallback, it an Attribute is not present.

Reply:
Yes, but the intentionality of the code is that `[JsonPropertyName]` trumps everything, regardless of the setting.  This is the behavior I want, and the code should read the same.